lib/renaissance/auctions/auction.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Renaissance.Auction do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  @required_fields ~w(title description user_id price end_date)a
+  @optional_fields ~w()a
+
+  schema "auctions" do
+    field :title, :string
+    field :description, :string
+    field :price, Money.Ecto.Amount.Type
+    field :end_date, :utc_datetime
+    belongs_to :user, User
+
+    timestamps(type: :utc_datetime)
+  end
+
+  @doc false
+  def changeset(user, attrs \\ %{}) do
+    user
+    |> cast(attrs, @required_fields ++ @optional_fields)
+    |> validate_required(@required_fields)
+    |> validate_price()
+    |> validate_date()
+  end
+
+  defp validate_price(changeset) do
+    item_price = get_change(changeset, :price)
+    zero_dollars = Money.new(000, :USD)
+    one_dollar = Money.new(100, :USD)
+
+    if Money.compare(item_price || one_dollar, zero_dollars) == 1 do
```

Comment:
Looking at this, it isn't clear what it means IMO

Reply:
I mean the comparison statement, could you extract it out?

Reply:
Resolved with 159a775e493da7136d042a4f76b690b4d8cb2aea